docs(fundamentals): updates testing to explain new setLogger method (#1633)
nestjs/docs.nestjs.com: content/fundamentals/unit-testing.md

In this example, we also provide an alternate (test-double) implementation of the `CatsService` which simply returns a hard-coded value that we can test for. Use `overrideProvider()` to provide such an alternate implementation. Similarly, Nest provides methods to override guards, interceptors, filters and pipes with the`overrideGuard()`, `overrideInterceptor()`, `overrideFilter()`, and `overridePipe()` methods respectively.

> 

Reviewer: ```suggestion
```

Each of the override method types, in turn, returns the `TestingModule` instance, and can thus be chained with other methods in the [fluent style](https://en.wikipedia.org/wiki/Fluent_interface). You should use `compile()` at the end of such a chain to cause Nest to instantiate and initialize the module.

> While testing you sometimes also want to provide a custom logger e.g. when the tests are run on a CI server. Use `setLogger()` with a `Logger` instance that is derived from the 

Reviewer: ```suggestion
Also, sometimes you may want to provide a custom logger e.g. when the tests are run (for example, on a CI server). Use the `setLogger()` method and pass an object that fulfills the `LoggerService` interface to instruct the `TestModuleBuilder` how to log during tests (by default, only "error" logs will be logged to the console).
```

> [Logger class](https://docs.nestjs.com/techniques/logger#injecting-a-custom-logger) to tell the TestModuleBuilder how to log during tests.

Reviewer: ```suggestion
```
